gtpv2/ie/ip-addr.go

```diff
@@ -165,7 +165,7 @@ func (i *IE) IPv6() (net.IP, error) {
 	}
 
 	switch i.Type {
-	case IPAddress, PDNAddressAllocation, S103PDNDataForwardingInfo, S1UDataForwarding:
+	case IPAddress, S103PDNDataForwardingInfo, S1UDataForwarding:
 		ip, err := i.IP()
```

Comment:
but IP already has all that you've changed. 
```
func (i *IE) IP() (net.IP, error) {
...
	switch i.Type {
...
	case PDNAddressAllocation:
...
		paa, err := ParsePDNAddressAllocationFields(i.Payload)
		if err != nil {
			return nil, err
		}
		switch paa.PDNType {
		case pdnTypeIPv4, pdnTypeIPv4v6:
			return paa.IPv4Address, nil
		case pdnTypeIPv6:
			return paa.IPv6Address, nil
...
```

Perhaps you need to update IP() parser

Reply:
My case is pdnTypeIPv4v6 and need to get ipv4 and ipv6. And how I can understand what version I need using IP()?